src/OpenLoco/src/Vehicles/Vehicle.cpp

```diff
@@ -686,6 +686,90 @@ namespace OpenLoco::Vehicles
         return AirportObjectFlags::acceptsHeavyPlanes;
     }
 
+    // 0x004AFFF3
+    VehicleBogie* VehicleBogie::flipCar()
+    {
+        Vehicle train(head);
+        VehicleBase* precedingCarBody = train.veh2;
+        while (precedingCarBody->nextVehicleComponent()->asBase<VehicleBogie>() != this)
+        {
+            precedingCarBody = precedingCarBody->nextVehicleComponent();
+        }
+
+        VehicleBase* ptr = this; // lvalue required
+        CarComponent firstComponent(ptr);
+        ptr = this; // CarComponent constructor changes value
+        Car car(ptr);
+
+        sfl::static_vector<CarComponent, VehicleObject::kMaxCarComponents> components;
+        for (CarComponent& component : car)
+        {
+            auto frontSprite = component.front->objectSpriteType;
+            component.front->objectSpriteType = component.back->objectSpriteType;
+            component.back->objectSpriteType = frontSprite;
+
+            // silence -Werror=null-dereference
+            auto body = component.body->asBase<VehicleCommon>();
+            if (body == nullptr)
+            {
+                return this;
+            }
+            body->var_38 ^= Flags38::isReversed;
+
+            components.push_back(component);
+        }
+        auto lastComponentIndex = components.size() - 1;
+        auto lastComponent = components[lastComponentIndex];
+
+        // if the Car is only one CarComponent we don't have to swap any values
+        if (lastComponent.front == firstComponent.front)
+        {
+            return this;
+        }
+
+        // silence -Werror=null-dereference
+        auto nextVehicleComponent = lastComponent.body->nextVehicleComponent();
+        if (nextVehicleComponent == nullptr)
+        {
+            return this;
+        }
+
+        precedingCarBody->nextEntityId = lastComponent.front->id;
+        firstComponent.body->nextEntityId = nextVehicleComponent->id;
+
+        for (; lastComponentIndex > 0; lastComponentIndex--)
+        {
+            components[lastComponentIndex].body->nextEntityId = components[lastComponentIndex - 1].front->id;
+        }
+
+        lastComponent.body->primaryCargo = firstComponent.body->primaryCargo;
+        lastComponent.body->breakdownFlags = firstComponent.body->breakdownFlags;
+        lastComponent.body->breakdownTimeout = firstComponent.body->breakdownTimeout;
+        lastComponent.front->secondaryCargo = this->secondaryCargo;
+        lastComponent.front->breakdownFlags = this->breakdownFlags;
+        lastComponent.front->breakdownTimeout = this->breakdownTimeout;
+        lastComponent.front->var_52 = this->var_52;
+        lastComponent.front->reliability = this->reliability;
+        lastComponent.front->breakdownTimeout = this->breakdownTimeout;
```

Comment:
this should be `timeoutToBreakdown` should really come up with a more unique name since it is easily confused with `breakdownTimeout`